libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/auth/RustMatrixAuthenticationService.kt

```diff
@@ -155,8 +157,8 @@ class RustMatrixAuthenticationService @Inject constructor(
                         passphrase = pendingPassphrase,
                         sessionPaths = currentSessionPaths,
                     )
-                clear()
                 sessionStore.storeData(sessionData)
+                authenticationObserver.onClientAuthenticationSucceeded(rustMatrixClientFactory.create(client))
```

Comment:
I don't really mind, and it seems like it could avoid some rare race conditions. Let's do it!

Reply:
Yes, since the insertion in the session store will trigger the switch of Appyx Node, it's probably better that the `matrixClient` is ready to be used before.